diesel/src/types/impls/date_and_time/chrono.rs

```diff
@@ -100,4 +134,57 @@ mod tests {
             .unwrap().nth(0).unwrap()
         );
     }
+
+    #[test]
+    fn times_of_day_encode_correctly() {
+        let connection = connection();
+
+        let midnight = NaiveTime::from_hms(0, 0, 0);
+        assert!(
+            connection.query_sql_params::<Bool, bool, Time, NaiveTime>(
+                "SELECT $1::time = '00:00:00'::time", &midnight)
+            .unwrap().nth(0).unwrap()
+        );
+
+        let noon = NaiveTime::from_hms(12, 0, 0);
+        assert!(
+            connection.query_sql_params::<Bool, bool, Time, NaiveTime>(
+                "SELECT $1::time = '12:00:00'::time", &noon)
+            .unwrap().nth(0).unwrap()
+        );
+
+        let roughly_half_past_eleven = NaiveTime::from_hms_micro(23, 37, 04, 2200);
+        assert!(
+            connection.query_sql_params::<Bool, bool, Time, NaiveTime>(
+                "SELECT $1::time = '23:37:04.002200'", &roughly_half_past_eleven)
+            .unwrap().nth(0).unwrap()
+        );
+
+    }
+
+    #[test]
+    fn times_of_day_decode_correctly() {
+        let connection = connection();
+        let midnight = NaiveTime::from_hms(0, 0, 0);
+        let midnight_from_sql = connection.query_sql::<Time, NaiveTime>("SELECT '00:00:00'::time")
+                                          .unwrap()
+                                          .nth(0)
+                                          .unwrap();
+        assert_eq!(midnight, midnight_from_sql);
+
+        let noon = NaiveTime::from_hms(12, 0, 0);
+        let noon_from_sql = connection.query_sql::<Time, NaiveTime>("SELECT '12:00:00'::time")
+                                      .unwrap()
+                                      .nth(0)
+                                      .unwrap();
+        assert_eq!(noon, noon_from_sql);
+
+        let roughly_half_past_eleven = NaiveTime::from_hms_micro(23, 37, 04, 2200);
+        let roughly_half_past_eleven_from_sql = connection.query_sql::<Time, NaiveTime>("SELECT '23:37:04.002200'::time")
```

Comment:
Can we indent these similarly to the other lines (and move the rhs to a newline as well)? These will just get replaced once #97 is fixed